### PR TITLE
Initialization for parameters

### PR DESCRIPTION
**Pull request by mbannick**
This did not work, at all. But putting it here for discussion!

### CODE REVIEW

File: src/sfma/models/marginal.py [hunk 1]
```diff
@@ -91,6 +92,52 @@ def objective(self, x: ndarray, data: Data) -> float:
...
+        v = data.obs_var + v_re + v_ie
+
+        z = np.sqrt(v_ie) * r / np.sqrt(2.0 * v * (data.obs_var + v_re))
+        x = self.femat
+
+        # Derivative of log erfc
+        dlerf = -(1 / erfc(z)) * 2 / np.sqrt(np.pi) * np.exp(-z**2)
+        grad = np.zeros(beta.size + 2)
+
+        for i in np.arange(data.obs.shape[0]):
```
Comment (zhengp0) [inline src/sfma/models/marginal.py]: We could talk about vectorize this for-loop to speed things up :)

File: src/sfma/models/marginal.py [hunk 2]
```diff
@@ -125,10 +172,28 @@ def get_var_init(self, data) -> ndarray:
...
+        # Calculate the first moment
+        # E[r_i] = E[u_i] + E[v_i] + E[\epsilon_i] + \alpha
+        # This expression is our estimate of \sqrt{2\eta/\pi}
+        eta = (np.mean(r) - alpha) ** 2 * np.pi / 2
+
+        # Calculate the second moment
+        # (\sum E[r_i^2] - \sum \sigma_i**2)/n =
+        #   \gamma + \eta (1 - 2/\pi) + (\alpha + \sqrt{2\eta / \pi})^2
+        moment2 = np.sum(r**2 - data.obs_se) / n
+        gamma = moment2 - eta * (1 - 2 / np.pi) - (alpha + np.sqrt(2 * eta / np.pi)) ** 2
```
Comment (zhengp0) [inline src/sfma/models/marginal.py:192]: great! interesting to see how it works.

**Review by zhengp0 — APPROVED** [review]
Thanks @mbannick! Looks good to me

File: src/sfma/models/marginal.py (hunk 2 above)
Comment (mbannick) [reply src/sfma/models/marginal.py:192]: it doesn't make any difference unfortunately... we will discuss!

**Comment by mbannick** [comment]
Thank you very much @zhengp0 !

**Pull request merged**